### PR TITLE
Updated tests not to exepct errors on hardware capabilities

### PR DESCRIPTION
**Pull request by DmitryVasilevsky**
This corresponds to https://github.com/microsoft/qsharp-compiler/pull/1595

### CODE REVIEW

File: src/Tests/AzureClientTests.cs [hunk 1]
```diff
@@ -404,7 +404,7 @@ public async Task TestRuntimeCapabilities(string targetId, AzureClientError? exp
             Assert.IsNotNull(azureWorkspace);
             azureWorkspace?.AddProviders("ionq", "quantinuum");
 
-            // Verify that IonQ job fails to compile (QPRGen0)
+            // Verify that IonQ job should generate warnings (QPRGen0)
```
Comment (samarsha) [inline src/Tests/AzureClientTests.cs:407]: It's really just verifying that submission succeeds, there's no assertion for warnings.

**Review by samarsha — COMMENTED** [review]
For VS Code/CLI development, we have the project templates to turn the warnings back into errors for some targets. But for IQ#, it looks like they're always warnings. Are we OK with that difference?

File: src/Tests/AzureClientTests.cs (hunk 1 above)
Comment (DmitryVasilevsky) [reply src/Tests/AzureClientTests.cs]: Updated comment.

**Comment by DmitryVasilevsky** [comment]
> For VS Code/CLI development, we have the project templates to turn the warnings back into errors for some targets. But for IQ#, it looks like they're always warnings. Are we OK with that difference?

I believe this was the goal (after discussion with Cesar).

**Review by samarsha — APPROVED** [review]

**Pull request merged**